Mathlib/Analysis/Fourier/AddCircleMulti.lean

```diff
@@ -198,7 +198,6 @@ def mFourierBasis : HilbertBasis (d → ℤ) ℂ L²(UnitAddTorus d) :=
 
 /-- The elements of the Hilbert basis `mFourierBasis` are the functions `mFourierLp 2`, i.e. the
 monomials `mFourier n` on `UnitAddTorus d` considered as elements of `L²`. -/
-@[simp]
```

Comment:
I think the  problem here is actually
```
instance instCoeFun : CoeFun (HilbertBasis ι 𝕜 E) fun _ => ι → E where
  coe b i := b.repr.symm (lp.single 2 i (1 : 𝕜))
```
which should be using `FunLike` instead.

Reply:
Or perhaps the coercion isn't injective, in which case probably this should be a new `HilbertBasis.coe` def.

Reply:
Eric, as mentioned above, this is being fixed in #21405

Reply:
I think #21440 is a better fix for `HilbertBasis` specifically.

Reply:
Yael has fixed this elsewhere.